api/specs.proto

```diff
@@ -333,6 +333,12 @@ message ContainerSpec {
 	//
 	// https://docs.docker.com/engine/reference/commandline/run/#configure-namespaced-kernel-parameters-sysctls-at-runtime
 	map<string, string> sysctls = 26;
+
+	// Swap limit equal to memory plus swap: '-1' to enable unlimited swap
+	int64 memory_swap = 27;
```

Comment:
why do we not need google.protobuf.Int64Value for this field?

Reply:
Because that value only needs to be an int64, not a *int64, https://github.com/moby/moby/blob/03e089e169a2889f096aceac6ad4a4e2ed6e186b/api/types/container/host_config.go#L333-L334

Reply:
right. if I don't want to change previously set memory swap value, how do I specify that?
Basically, why does memory_swappiness need to be nullable but memory_swap does not?

Reply:
"if I don't want to change previously set memory swap value, how do I specify that?" => I think that's the same for all the params here:
* moby's API does expect a full service spec for both creates and updates, you can't just `PATCH` an existing service
* if your services are created/updated using docker-compose.yml file, then you have all the params in there, and they will all be around when updating
* if you use the CLI, then it's responsible for fetching the current service spec and merging with the new params (https://github.com/docker/cli/blob/master/cli/command/service/update.go#L239-L482); arguably it would be nice to have the API handle that eventually?

So in particular, memory_swappiness doesn't _need_ to be nullable... It just seems good to be consistent with moby's container spec struct?

Reply:
ok cool lets move forward as is for consistency.